--- a/code/SequencesLoader/SequencesLoader.cpp	2013-04-17 09:44:55.000000000 -0400
+++ b/code/SequencesLoader/SequencesLoader.cpp	2013-08-27 10:09:56.168561858 -0400
@@ -201,6 +201,9 @@
 	m_loader.constructor(m_parameters->getMemoryPrefix().c_str(),m_parameters->showMemoryAllocations(),
 		m_rank);
 
+	char * seq = __Malloc(RAY_MAXIMUM_READ_LENGTH * sizeof(char));
+	char * qlt = __Malloc(RAY_MAXIMUM_READ_LENGTH * sizeof(char));
+
 	for(m_distribution_file_id=0;m_distribution_file_id<(int)allFiles.size();
 		m_distribution_file_id++){
 
@@ -212,11 +215,9 @@
 
 		// write Reads in AMOS format.
 		if(rank==MASTER_RANK&&m_parameters->useAmos()){
-			char qlt[20000];
 			for(LargeIndex i=0;i<m_loader.size();i++){
 				ReadHandle iid=m_distribution_currentSequenceId;
 				m_distribution_currentSequenceId++;
-				char seq[4000];
 				m_loader.at(i)->getSeq(seq,m_parameters->getColorSpaceMode(),true);
 				#ifdef ASSERT
 				assert(seq!=NULL);
@@ -237,6 +238,10 @@
 		}
 		m_loader.reset();
 	}
+
+	__Free(seq);
+	__Free(qlt);
+
 	m_loader.clear();
 	if(m_parameters->useAmos()){
 		fclose(fp);
